erdpy/proxy/core.py

```diff
@@ -129,7 +129,9 @@ def send_transaction_and_wait_for_result(self, payload: Any, num_seconds_timeout
 
         for _ in range(0, num_seconds_timeout):
             if self.is_transaction_finalized(tx_hash):
-                return self.get_transaction(tx_hash=tx_hash, with_results=True)
+                tx = self.get_transaction(tx_hash=tx_hash, with_results=True)
+                tx['hash'] = tx_hash
```

Comment:
It might be better if `tx['hash'] = tx_hash` were moved into `self.get_transaction()` itself, at lines 113/114. Then the hash would always be set. Strange that the proxy doesn't set the tx hash in the response... (?)